tests/utils/network_setup.go

```diff
@@ -0,0 +1,270 @@
+			{
+				VmName:      evm.IDStr,
+				Genesis:     warpGenesisFile,
+				ChainConfig: warpChainConfigPath,
+				SubnetSpec: &rpcpb.SubnetSpec{
+					SubnetConfig: "",
+					Participants: SubnetANodeNames,
+				},
+			},
+			{
+				VmName:      evm.IDStr,
+				Genesis:     warpGenesisFile,
+				ChainConfig: warpChainConfigPath,
+				SubnetSpec: &rpcpb.SubnetSpec{
+					SubnetConfig: "",
+					Participants: SubnetBNodeNames,
+				},
+			},
+		},
+	)
+	Expect(err).Should(BeNil())
+
+	// Issue transactions to activate the proposerVM fork on the chains
+	FundedKey, err = crypto.HexToECDSA(FundedKeyStr)
+	Expect(err).Should(BeNil())
+	SetUpProposerVM(ctx, FundedKey, manager, 0)
+	SetUpProposerVM(ctx, FundedKey, manager, 1)
+
+	// Set up subnet URIs
+	SubnetIDs = manager.GetSubnets()
+	Expect(len(SubnetIDs)).Should(Equal(2))
+
+	SubnetA = SubnetIDs[0]
+	SubnetADetails, ok := manager.GetSubnet(SubnetA)
+	Expect(ok).Should(BeTrue())
+	Expect(len(SubnetADetails.ValidatorURIs)).Should(Equal(5))
+	BlockchainIDA = SubnetADetails.BlockchainID
+	ChainANodeURIs = append(ChainANodeURIs, SubnetADetails.ValidatorURIs...)
+
+	SubnetB = SubnetIDs[1]
+	SubnetBDetails, ok := manager.GetSubnet(SubnetB)
+	Expect(ok).Should(BeTrue())
+	Expect(len(SubnetBDetails.ValidatorURIs)).Should(Equal(5))
+	BlockchainIDB = SubnetBDetails.BlockchainID
+	ChainBNodeURIs = append(ChainBNodeURIs, SubnetBDetails.ValidatorURIs...)
+
+	log.Info(
+		"Created URIs for subnets",
+		"ChainAURIs", ChainANodeURIs,
+		"ChainBURIs", ChainBNodeURIs,
+		"BlockchainIDA", BlockchainIDA,
+		"BlockchainIDB", BlockchainIDB,
+	)
+
+	chainAWSURI := HttpToWebsocketURI(ChainANodeURIs[0], BlockchainIDA.String())
+	ChainARPCURI = HttpToRPCURI(ChainANodeURIs[0], BlockchainIDA.String())
+	log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI, "rpcURL, ChainARPCURI")
+	ChainAWSClient, err = ethclient.Dial(chainAWSURI)
+	Expect(err).Should(BeNil())
+	ChainARPCClient, err = ethclient.Dial(ChainARPCURI)
+	Expect(err).Should(BeNil())
+
+	ChainAIDInt, err = ChainARPCClient.ChainID(context.Background())
+	Expect(err).Should(BeNil())
+
+	chainBWSURI := HttpToWebsocketURI(ChainBNodeURIs[0], BlockchainIDB.String())
+	ChainBRPCURI = HttpToRPCURI(ChainBNodeURIs[0], BlockchainIDB.String())
+	log.Info("Creating ethclient for blockchainB", "wsURI", chainBWSURI)
+	ChainBWSClient, err = ethclient.Dial(chainBWSURI)
+	Expect(err).Should(BeNil())
+	ChainBRPCClient, err = ethclient.Dial(ChainBRPCURI)
+	Expect(err).Should(BeNil())
+
+	ChainBIDInt, err = ChainBRPCClient.ChainID(context.Background())
+	Expect(err).Should(BeNil())
+
+	NewHeadsA = make(chan *types.Header, 10)
+
+	log.Info("Finished setting up e2e test subnet variables")
+}
+
+// DeployTeleporterContract deploys the Teleporter contract to the two subnets. The caller is responsible for generating the
+// deployment transaction information
+func DeployTeleporterContract(transactionBytes []byte, deployerAddress common.Address, contractAddress common.Address) {
```

Comment:
I think if we refactor this to take a slice of RPC clients as a param, and then loop over them to perform the deploy, we can simplify this function a lot. Also if we execute the fund/deploy on both chains before sleeping, we can speed up the tests a little bit by eliminating one of the sleeps.

Reply:
Implemented this. Thanks for the suggestion.